src/circuit/circuit_frame.rs

```diff
@@ -1388,6 +1408,25 @@ fn reduce_cons<F: LurkField, CS: ConstraintSystem<F>>(
     let head_is_current_env = and!(cs, &head_is_current_env0, &head_is_a_sym)?;
     let head_is_if = and!(cs, &head_is_if0, &head_is_a_sym)?;
 
+    let mut head_is_coprocessor_bools = Vec::with_capacity(lang.coprocessors().len());
+
+    for (sym, (_coproc, scalar_ptr)) in lang.coprocessors().iter() {
+        let cs = &mut cs.namespace(|| format!("head is {}", sym.full_name()));
+
+        let allocated_boolean = head.alloc_hash_equal(cs, *scalar_ptr.value())?;
+
+        head_is_coprocessor_bools.push(allocated_boolean);
+    }
+
+    let head_is_coprocessor = if head_is_coprocessor_bools.is_empty() {
+        Boolean::Constant(false)
+    } else {
+        constraints::or_v_unchecked_for_optimization(
```

Comment:
Why do we need this "unchecked for optimization"?

Reply:
Because if you have an alternative between less than 4 coprocs here:
https://github.com/lurk-lab/lurk-rs/blob/db62b0be6a1ec279921e392f04ca9cb7afc5c39e/src/circuit/gadgets/constraints.rs#L572-L575
Then you will be barfed at. And in the current setup (one or no coprocessors, depending on testing setup), that would happen.